src/t/d11.clj

```diff
@@ -0,0 +1,88 @@
+(ns t.d11
+  (:gen-class)
+  (:require
+    [clojure.string :as cljstr]
+    [clojure.set :as cljset]
+    [clojure.core.match :refer [match]]))
+
+(def sample
+"125 17")
+
+(defn parse-input
+  [input]
+  (->> input
+       (re-seq #"\d+")
+       (map parse-long)
+  ))
+
+(defn split-stone-2
+  [stone]
+  (let [str-stone (str stone)
+        mid-i (quot (count str-stone) 2)]
+    (->>
+      [(take mid-i str-stone) (drop mid-i str-stone)]
+      (map (fn [l] (apply str l)))
+      (map parse-long))))
+
+(defn evolve-stone
+  [stone]
+  (cond
+   (= stone 0) [1]
+   (even? (count (str stone))) (split-stone-2 stone)
+   :else [(* stone 2024)]))
+
+(def memo-evolve-stone (memoize evolve-stone))
```

Comment:
I don't think this function is expensive enough for memoization to help.

Reply:
yeah, we discussed it, I understand that now, might not remember it for next year but learning goes through repetition right ?